### ISSUE TITLE
rebuildOnUpdate looses Swiper reference

### ISSUE THREAD

**Issue by jozsefDevs** [kidjp85/react-id-swiper#438]
Hey all! 

**Problem**: When using Swiper with rebuildOnUpdate=true I am loosing reference to Swiper instance. This becomes a problem because as time goes on I need to replace all slides in the swiper. When replacement done, I will loose connection to 'slideChange' event totally.

**Demo**: https://codesandbox.io/s/react-id-swiper-update-problem-q590f

**Steps to reproduce**:
Step: 1 - Swiper loads in with Dataset1
Step: 2 - Open up console, swipe some slides to left/right, see console.logs working
Step: 3 - Hit toggle dataset
Step: 4 - Reference to Swiper instance gone (no more logs on slide change)

***Possible connections***: #327

**Comment by Magtheridon96** [comment]
I left a comment on the possibly connected issue.
I actually needed to sprinkle in some calls to `getSwiperInstance(swiperInstanceRef.current);` to hack around it.

I think this is because the changes to the `ref` are treated as post-render side-effects, so the component enters an undesirable state where the `ref` is up to date internally, but parents still hold a `ref` to a destroyed swiper. For this reason (and actually, even after adding the extra calls to `getSwiperInstance`, I had to check `swiper.destroyed` in whatever callbacks I had that involved the swiper.

I don't know what is up with this library, but there are far too many problems...
